tests/unit/src/test/scala/tests/FormattingSlowSuite.scala

```diff
@@ -225,4 +226,40 @@ object FormattingSlowSuite extends BaseSlowSuite("formatting") {
     } yield ()
   }
 
+  testAsync("missing-version") {
+    cleanWorkspace()
+    for {
+      _ <- server.initialize(
+        """|/.scalafmt.conf
+           |maxColumn=40
+           |/Main.scala
+           |object   Main
+           |""".stripMargin,
+        expectError = true
+      )
+      _ <- server.didOpen("Main.scala")
+      _ <- server.formatting("Main.scala")
+      _ = assertNoDiff(
+        client.workspaceMessageRequests,
+        MissingScalafmtVersion.messageRequestMessage
+      )
+      _ = assertNoDiff(
+        client.workspaceShowMessages,
+        MissingScalafmtVersion.fixedVersion.getMessage
+      )
+      _ = assertNoDiff(client.workspaceDiagnostics, "")
+      // check file was not formatted because version was missing.
+      _ = assertNoDiff(
+        server.bufferContent("Main.scala"),
+        "object   Main"
+      )
+      _ <- server.formatting("Main.scala")
+      // check file was formatted now that version has been added.
```

Comment:
Two things:

- I don't understand (reading the test), where is the interaction that adds the version
- I would also test against the content of `.scalafmt.conf`, which should now contain the version

Reply:
Good points, will update.